evaporate.js

```diff
@@ -314,10 +339,14 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
               } else {
                  part.status = ERROR;
                  part.loadedBytes = 0;
+
+                 awsResponse = getAwsResponse(xhr);
+                 if (awsResponse.code) {
+                    l.e('AWS Server response: code="' + awsResponse.code + '", message="' + awsResponse.msg + '"');
```

Comment:
@tomsaffell AWS returns a nice payload when it finds that the checksum mismatches. We might want to special-case that as it's pretty fundamental to what this project seems to be about.